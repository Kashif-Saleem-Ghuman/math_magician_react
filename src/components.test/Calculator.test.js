import React from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Calculator from '../components/Calculator';

afterEach(cleanup);

describe('Calculator', () => {
  test('should render without error', () => {
    render(<Calculator />);
  });

  test('should update state when button is clicked', () => {
    const { getByText } = render(<Calculator />);
    const button = getByText('7');
    fireEvent.click(button);
    const answer = '7';
    expect(answer).toBe('7');
  });

  test('should correctly perform calculations when buttons are clicked', () => {
    const { getByText } = render(<Calculator />);
    const button7 = getByText('7');
    const buttonX = getByText('x');
    const button3 = getByText('3');
    const buttonEqual = getByText('=');
    fireEvent.click(button7);
    fireEvent.click(buttonX);
    fireEvent.click(button3);
    fireEvent.click(buttonEqual);
    const answer = parseInt(button7.textContent, 10) * parseInt(button3.textContent, 10);
    expect(`${answer}`).toBe('21');
  });

  test('matches snapshot for new button', () => {
    const tree = renderer.create(<Calculator />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
