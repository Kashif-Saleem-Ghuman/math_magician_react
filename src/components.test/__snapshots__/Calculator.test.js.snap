// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator matches snapshot for new button 1`] = `
<div
  className="calculator"
>
  <h2>
    Lets do some Math!
  </h2>
  <div>
    <div
      className="answer"
    >
      0
    </div>
    <div
      className="operations"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="maths"
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="maths"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="maths"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="maths"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
      <button
        className="zero"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="maths"
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;
